test/puppetlabs/puppetdb/scf/storage_test.clj

```diff
@@ -135,14 +144,72 @@
                               set)]
         (is (= stored-names (set (keys facts))))))))
 
+(deftest fact-value-row-changes-basic-behavior
+  (let [changes fact-value-row-changes
+        value-row (fn [x hash]
+                    (let [type-id (fact-value-type-id x)]
+                      [type-id
+                       hash
+                       (sutils/munge-jsonb-for-storage x)
+                       (when (zero? type-id) x)      ; value_string
+                       (when (= 1 type-id) x)        ; value_integer
+                       (when (= 2 type-id) x)        ; value_float
+                       (when (= 3 type-id) x)        ; value_boolean
+                       ]))]
+    (is (not (apply changes "x" (value-row "x" nil))))
```

Comment:
OK - I'd been a little inclined against `(are ...)` because I thought it presented the wrong line numbers on failures, and if so, in simple cases, didn't mind the copy/pasting, but I'm happy to change it if y'all prefer.